jetstream/analysis.py

```diff
@@ -229,6 +235,18 @@ def run(self, current_date: datetime, dry_run: bool) -> None:
 
         for period in self.config.metrics:
             time_limits = self._get_timelimits_if_ready(period, current_date)
+
+            if dry_run:
+                # make sure time_limits has some value so that dry run starts running
+                time_limits = TimeLimits.for_single_analysis_window(
+                    first_enrollment_date=self.config.experiment.start_date.strftime("%Y-%m-%d"),
+                    last_date_full_data=self.config.experiment.end_date.strftime("%Y-%m-%d"),
+                    analysis_start_days=0,
+                    analysis_length_dates=(
+                        self.config.experiment.end_date - self.config.experiment.start_date
+                    ).days,
+                )
```

Comment:
If we're doing "dry run" and not "query validation" then I don't think we want this

Reply:
My reasoning was to make sure the query gets generated and dry run for every `period`. Otherwise `time_limits` will depend on the date the dry run is performed. So in some days weekly or overall queries might not be dry run.

Reply:
Okay, I see. Maybe I'm just hung up on semantics; "dry run" means "do exactly the work you would have done but don't change anything on disk" to me. We don't actually want that behavior. If we called this "validation mode" and wrote down what that meant I'd support this (and it addresses my concerns about "no, you have to run this for every day the experiment will be active").

Reply:
Also I think we can just pass analysis_length_dates=1 here because it's not like we care 👼

Reply:
You are right. If we run `rerun --dry_run` then we cover all periods and and dates in the dry run. I'll remove this part here.